google/cloud/spanner/value.cc

```diff
@@ -88,6 +90,52 @@ bool Equal(google::spanner::v1::Type const& pt1,
   }
 }
 
+// A 256-element table mapping all `unsigned char` byte values to a printable
+// string. All naturally printable values are used directly. All non-printable
+// bytes use their octal escape sequence. Whitespace characters use their short
+// form like '\t' and '\n'.
+using ByteType = unsigned char;
+constexpr auto kByteTableSize = std::numeric_limits<ByteType>::max() + 1;
+constexpr std::array<char const*, kByteTableSize> kByteTable = {
+    "\\000", "\\001", "\\002", "\\003", "\\004", "\\005", "\\006", "\\a",
```

Comment:
Do we care if the application is using a non-ASCII locale?

Reply:
Maybe. Can you say more about what that'd affect here? Is there something different you'd recommend?

Reply:
So, it looks like we could handle other locales if we compute the "isprint" at runtime. That means, we'd run a little code instead of using a lookup table. That'd be fine with me. Do you have a preference?

Reply:
I rolled back to an earlier version of this PR where I use `std::isprint` at runtime, which is what I believe you'res suggesting. I think this should now work with non-ASCII locales.